header.go

```diff
@@ -1473,7 +1473,7 @@ func (h *RequestHeader) setSpecialHeader(key, value []byte) bool {
 			return true
 		case caseInsensitiveCompare(strCookie, key):
 			h.collectCookies()
-			h.cookies = parseRequestCookies(h.cookies, value)
+			h.cookies = parseRequestCookies(make([]argsKV, 0), value)
```

Comment:
The setSpecialHeader method is used in both the Set and Add methods of Header, so modifying it will affect its usage in Add. To reduce code duplication, an additional parameter can be added to setSpecialHeader, or h.Cookies can be reset before calling setSpecialHeader based on the parent method. Resetting h.Cookies can be done using RequestHeader.DelAllCookies's zero slice strategy, as using make may result in unnecessary memory allocation.

Such a pull request may not necessarily be accepted by @erikdubbelboer, but personally, I would accept it. Why not give it a try? 😊